modules/dnn/test/test_common.hpp

```diff
@@ -137,7 +136,6 @@ bool validateVPUType();
 
 testing::internal::ParamGenerator< tuple<Backend, Target> > dnnBackendsAndTargets(
         bool withInferenceEngine = true,
-        bool withHalide = false,
```

Comment:
Might break the tests. Can we just check for `CV_Assert(!withHalide`)?

Reply:
I can review every location where `dnnBackendsAndTargets` is used, and make changes if needed. Let me try.

Reply:
All `dnnBackendsAndTargets` called with parameters should be taken well care now.

Reply:
Sounds good, thanks

Reply:
> I can review every location

This is not one time activity.
In case of parameter change we need to review each of "(5.x) Merge 4.x" PRs.
It is much safer to rename this parameter to `bool obsolete_withHalide = false`.

Reply:
So how can we completely remove that in this case?

Reply:
I think we can resolve this thread. The latest commits provide compatible removal of `withHalide` flag from `dnnBackendsAndTargets`.

Reply:
Again, fixing it here will not resolve merging  4.x->5.x issues (no errors would be raised in the most part of cases) which would lead to mistakes.

Reply:
So how about removing the parameter from 4.x as well like what we want to do with `modules/dnn/test/test_halide_layers.cpp`

Reply:
So how do we do for this issue?

Reply:
Series of boolean parameters is anti-pattern, especially in form with optional default values.
Any obvious refactoring is dangerous here, especially in case of regular merges between branches.

> It is much safer to rename this parameter to `bool obsolete_withHalide = false`

Reply:
> It is much safer to rename this parameter to bool obsolete_withHalide = false

Okay, I will do this. I guess we can safely remove this legacy once our main dev branch switches to 5.x, right?